lib/api.rb

```diff
@@ -157,19 +218,31 @@ def validation_error(msg)
         return validation_error("Unknown error")
       end
 
+      headers "access-control-allow-origin" => "*"
+
       {
         :access_token => d.access_token,
         :expires_in => (d.token_expires_at - Time.now).floor,
         :token_type => "Bearer",
         :refresh_token => d.refresh_token,
         :Key => d.user.key,
+        :PrivateKey => d.user.private_key,
         # TODO: when to include :privateKey and :TwoFactorToken?
       }.to_json
     end
   end
 end
 
 namespace BASE_URL do
+
+  # For HTTP CORS verification
+  options "*" do
+    response.headers["Allow"] = "GET, POST, OPTIONS, PUT, DELETE"
+    response.headers["Access-Control-Allow-Headers"] = "Authorization, Content-Type, Accept"
+    response.headers["Access-Control-Allow-Origin"] = "*"
+    200
+  end
+
```

Comment:
not needed if web-vault is hosted under the same domain as the backend, which I guess would be the usual setup.

Reply:
You’re right. But it need a reverse proxy in front (that’s what i did).